core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1143,10 +1143,10 @@ private static void verifyChecksum(BufferedChecksumStreamInput in) throws IOExce
             }
             operation = Translog.Operation.readType(in);
             verifyChecksum(in);
+        } catch (TranslogCorruptedException e) {
+            throw e;
         } catch (EOFException e) {
             throw new TruncatedTranslogException("reached premature end of file, translog is truncated", e);
-        } catch (AssertionError | Exception e) {
-            throw new TranslogCorruptedException("translog corruption while reading from stream", e);
```

Comment:
I agree with the reasoning. So we just need to check we have no code that has special handling of `TranslogCorruptedException`s (maybe you already checked)?